File: workout-api/src/main/java/tom/projects/monkeylog/model/exercise/Exercise.java
package tom.projects.monkeylog.model.exercise;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import tom.projects.monkeylog.model.user.UserOwned;
import tom.projects.monkeylog.model.workout.ExerciseGroup;

import java.util.List;

@Entity
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class Exercise implements UserOwned {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private Long userId;
    private String name;
    @OneToMany(mappedBy = "exercise")
    private List<ExerciseGroup> exerciseGroups;
    @ManyToOne
    @JoinColumn(name = "exercise_type_id", nullable = false)
    private ExerciseType exerciseType;
}

File: backend/src/main/kotlin/tom/com/workout/logbook/model/exercise/Exercise.kt
package tom.com.workout.logbook.model.exercise

import jakarta.persistence.*
import org.hibernate.annotations.UuidGenerator
import tom.com.workout.logbook.model.user.UserOwned
import tom.com.workout.logbook.model.workout.ExerciseGroup
import java.util.*

@Entity
class Exercise(
    @Id
    @GeneratedValue
    @UuidGenerator
    var id: UUID? = null,
    override var userId: UUID? = null,
    var name: String,
    @OneToMany(mappedBy = "exercise")
    var exerciseGroups: List<ExerciseGroup> = ArrayList(),
    @Enumerated(EnumType.STRING)
    var exerciseCategory: ExerciseCategory
) : UserOwned
